### ISSUE TITLE
E-Reader plugin can't properly compose EPUB files

### ISSUE THREAD

**Issue by caribou-slim** [akirk/friends-send-to-e-reader#10]
The generated E-PUB files are corrupt. Output provides an E-PUB that only displays the first post in the series when viewed in a standard reader and a table of contents with a blank link. Images from multiple posts exist within the EPUB file, but can't be displayed unless you use a Comic reader like CDisplayEx - which can't see any text whatsoever. Diving into the E-PUB with zip viewer shows that the information is being captured, but is somehow getting corrupted for the e-reader interface. 

Tested  on multiple E-PUB readers.

Adjusting the internal settings doesn't seem to have any impact. Would love for this to work, but have to currently disable it as it's non-functional.

**Comment by akirk** [comment]
Thanks for reporting and sorry for the trouble! I use it every day and have not experienced what you're seeing.
Would you be willing to share such a broken epub (from the wp-content/uploads folder)? 
Maybe a post contains incompatible HTML that I'm not considering. You could also try the list-URL to try with a different selection of posts to see if it is a specific post that trips it up.

**Comment by caribou-slim** [comment]
Thanks for the quick response!

Had to zip it up to get github to accept it, but here you go:
[Uploading caribou-slim-november-23-2024-1.zip.zip…]()

Here's a link to download yourself:
https://faewild.space/friends/caribou-slim-pagan-plus/?epubad5c382e=all

**Comment by akirk** [comment]
Hm, so if I try this with another user on the blog, like "caribou_slim" it produces a fine epub for me: https://faewild.space/friends/caribou_slim/?epubad5c382e=all

It looks like your friend with the slug "caribou-slim-pagan-plus" only has a single post without title (likely it's a status post) which is why it doesn't show up in the table of contents. Actually it does (see below), just with no text, I've fixed that in 0.8.2.

Could you update the plugin and check if it now works better? Thanks!


The post in the TOC:
```
<p class='level1'><a href="0.html"></a></p>
```

**Comment by caribou-slim** [comment]
Thanks again for your work on this - especially on the weekend!

Unfortunately, it looks like the issue is still in play on my end after the update.

It seems to have problems based on whether or not the author has a wordpress account on the domain or not. 

For instance, the first feed sample I gave to you comes from a mastodon account, and should be showing 15 statuses (which do appear on the Friends page):

https://faewild.space/friends/caribou-slim-pagan-plus/?epubad5c382e=all 

The successful feed you referenced is a separate account on the wordpress account itself:

https://faewild.space/friends/caribou_slim/?epubad5c382e=all

For a better example of how post displays aren't working for Friends without a Wordpress account on the instance, here's an example of another followed user via Mastodon:

https://faewild.space/friends/literary-pug/?epubad5c382e=all

- This friend (literarypub) is displaying 25 posts according to Friends, but only one shows up in the reader. 

Here's another example:

https://faewild.space/friends/aphelion/?epubad5c382e=all

- This friend (aphelion) has 37 posts and 12  statuses - only one shows up in the E-Reader. 

It does make sense that Statuses would have difficulty displaying due to a lack of title. However, the problem persists even with Posts.  

Just wanted to say please don't rush on this and enjoy your weekend! Thanks again for your great work - I love what you've done with Friends - groundbreaking work for the Fediverse and the Wordpress worlds.

**Comment by akirk** [comment]
Thank you for the many examples! I took another stab at this with #12, realizing that I didn't give it a title in the table of contents. Also, they would all get the same title (Untitled), so I now try to use the date so that the titles are different. Could you try again?

**Comment by caribou-slim** [comment]
Sorry it took me a bit to get back to you. Unfortunately looks like it's having the same issues after the update, so I've tried to track down the issues with the titling. I've reviewed the Cached Posts -> Posts from Friends. It appears that anything coming in the Friends plugin  from offsite users isn't generating a title, which I suspect is causing the TOC to fail. I'm going to enter some titles in those posts and see if it fixes the issue, if so, injecting a default title into Posts from Friends may do the trick. I'll send an update once I've had a chance to test.
